core/configuration.py

```diff
@@ -0,0 +1,131 @@
+        else:
+            sys.stderr.write("Error: No config.json file found.\n")
+            sys.exit(0)
+
+        return config
+
+    def _load_config_file(self, configfile):
+        """
+        Loads json from config file
+
+        :return config: A dictionary of configuration data.
+
+        """
+        try:
+            config = json.load(configfile)
+        except ValueError, e:
+            sys.stderr.write('Error: Config json file is invalid.\n')
+            print(e)
+            sys.exit(0)
+
+        return config
+
+    def _define_output_file(self, args):
+        """
+        Defines the filename for GTFS file to write.
+        Either from config file or command argument.
+
+        :return output: Filename for saving GTFS output file.
+
+        """
+        # Get and check filename for gtfs output
+        if args.output is not None:
+            output_file = args.output
+        elif 'output_file' in self.config:
+            output_file = self.config['output_file']
+        else:
+            sys.stderr.write('Error: No filename for gtfs file specified.\n')
+            sys.exit(0)
+
+        return output_file
+
+    def _prepare_dates(self):
+        """
+        Validate and prepare start and end date.
+        Either from config file or based on current date.
+
+        """
+        config = self.config
+
+        try:
+            start_date = datetime.datetime.strptime(
+                config['feed_info']['start_date'], "%Y%m%d")
+        except:
+            # Use first of current month if no start date was specified
+            now = datetime.datetime.now()
+            start_date = datetime.date(day=1, month=now.month, year=now.year)
+            config['feed_info']['start_date'] = str(start_date.year) + \
+                str(start_date.month) + "01"
+
+        print("Using the following start date: " +
+              config['feed_info']['start_date'])
+
+        try:
+            end_date = datetime.datetime.strptime(
+                config['feed_info']['end_date'], "%Y%m%d")
+        except:
+            # Define end date automatically one year from start date
+            end_date_year = start_date.year + 1
+            end_date_month = start_date.month - 1
+            end_date_day = monthrange(end_date_year, end_date_month)[1]
+            end_date = datetime.date(
+                day=end_date_day, month=end_date_month, year=end_date_year)
+            config['feed_info']['end_date'] = str(
+                end_date.year) + str(end_date.month) + str(end_date.day)
+
+        print("Using the following end date: " +
+              config['feed_info']['end_date'])
+
+        # Validate dates
+        if start_date > end_date:
+            sys.stderr.write('Error: End dates finishes before start date.\n')
+            sys.exit(0)
+        elif end_date - start_date > datetime.timedelta(days=364):
+            sys.stderr.write(
+                "Dates are describing range of more than one year.\n")
```

Comment:
Good that this a non-fatal warning as not everybody might want to use the created GTFS with navitia and other consumers might not have this one-year limitation. However, you could write `Warning: ` here like you wrote `Error: `, so it is immediately clear what status this message has.